docs/metrics.md

```diff
@@ -323,13 +513,117 @@ metrics.add_counter('LoopCount', -1, None, dimensions)
 
 ```
 
-### Log custom metrics
+### Getting a metric
+
+Users can [get a metric](https://github.com/pytorch/serve/tree/master/ts/metrics/metric_cache_yaml.py#L202)
+from the cache. The Metric object is returned, so the user can access the methods of the Metric:
+(i.e. `Metric.update(value)`, `Metric.__str__`)
+
+```python
+    def get_metric(self, metric_type: MetricTypes, metric_name: str, dimensions: list or str) -> Metric:
+        """
+        Get a Metric from cache.
+            Ask user for required requirements to form metric key to retrieve Metric.
+
+        Parameters
+        ----------
+        metric_type: MetricTypes
+            Type of metric: use MetricTypes enum to specify
+
+        metric_name: str
+            Name of metric
+
+        dimensions: list or str
+            list of dimension keys which should be strings
+
+        """
+```
+
+i.e.
+```python
+# Method 1: Getting metric of MetricType counter, metric name string, and list of dimension keys derived from YAML file.
+metrics.get_metric(MetricTypes.counter, "MetricName", ["ModelName", "Level"])
+
+# Method 2: Getting metric of MetricType gauge, metric name string, and complete log string of dimensions.
+metrics.get_metric(MetricTypes.gauge, "GaugeMetricName", "ModelName:foo,Level:Model")
+```
+
+
+## Log custom metrics
 
 Following sample code can be used to log the custom metrics created in the model's custom handler:
 
+_Metrics will only be emitted if they have value assigned to them.
+In other words, if metrics have an initial value of 0 (which is the default value), the metric will not be emitted._
+
 ```python
-for metric in metrics.store:
-    logger.info("[METRICS]%s", str(metric))
+# In Custom Handler
+from ts.service import emit_metrics
+
+class ExampleCustomHandler(BaseHandler, ABC):
+   def initialize(self, ctx):
+
+emit_metrics(metrics.cache)
 ```
 
 This custom metrics information is logged in the `model_metrics.log` file configured through [log4j2.xml](https://github.com/pytorch/serve/blob/master/frontend/server/src/main/resources/log4j2.xml) file.
+
+**NOTE:** After emitting the metrics, all metric values that have been updated will be reset to 0.
+
+## Metrics YAML File Parsing and Metrics API Custom Handler Example
+
+This example utilizes the feature of parsing metrics from a YAML file, adding and updating metrics and their values via Metrics API,
+updating metrics that have been parsed from the YAML file via Metrics API, and finally emitting all metrics that have been updated.
+
+```python
+from ts.service import emit_metrics
+from ts.metrics.metric_type_enum import MetricTypes
+
+
+class CustomHandlerExample:
```

Comment:
I tried packaging and running this example and still couldn't see any logs printed 

```
java.lang.InterruptedException: null
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.reportInterruptAfterWait(AbstractQueuedSynchronizer.java:2056) ~[?:?]
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2133) ~[?:?]
        at java.util.concurrent.ArrayBlockingQueue.poll(ArrayBlockingQueue.java:432) ~[?:?]
        at org.pytorch.serve.wlm.WorkerThread.run(WorkerThread.java:189) [model-server.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
2022-10-11T23:48:34,424 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_service_worker.py", line 120, in load_model
2022-10-11T23:48:34,424 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     service = model_loader.load(
2022-10-11T23:48:34,424 [WARN ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.BatchAggregator - Load model failed: metrics_model, error: Worker died.
2022-10-11T23:48:34,425 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_loader.py", line 135, in load
2022-10-11T23:48:34,425 [DEBUG] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - W-9000-metrics_model_1.0 State change WORKER_STARTED -> WORKER_STOPPED
2022-10-11T23:48:34,425 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     entry_point, initialize_fn = self._get_class_entry_point(module)
2022-10-11T23:48:34,425 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_loader.py", line 197, in _get_class_entry_point
2022-10-11T23:48:34,425 [WARN ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - terminateIOStreams() threadName=W-9000-metrics_model_1.0-stderr
2022-10-11T23:48:34,425 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     raise ValueError(
2022-10-11T23:48:34,425 [WARN ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - terminateIOStreams() threadName=W-9000-metrics_model_1.0-stdout
2022-10-11T23:48:34,425 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - ValueError: Expect handle method in class <class 'metrics_model.CustomHandlerExample'>
2022-10-11T23:48:34,426 [INFO ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - Retry worker: 9000 in 13 seconds.
2022-10-11T23:48:34,426 [INFO ] W-9000-metrics_model_1.0-stdout org.pytorch.serve.wlm.WorkerLifeCycle - Stopped Scanner - W-9000-metrics_model_1.0-stdout
2022-10-11T23:48:34,443 [INFO ] W-9000-metrics_model_1.0-stderr org.pytorch.serve.wlm.WorkerLifeCycle - Stopped Scanner - W-9000-metrics_model_1.0-stderr
2022-10-11T23:48:47,426 [DEBUG] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - Worker cmdline: [/opt/conda/envs/serve/bin/python3.8, /opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_service_worker.py, --sock-type, unix, --sock-name, /tmp/.ts.sock.9000, --metrics-config, /home/ubuntu/serve/frontend/server/src/test/resources/metrics_default.yaml]
2022-10-11T23:48:48,268 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - Listening on port: /tmp/.ts.sock.9000
2022-10-11T23:48:48,269 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - [PID]24436
2022-10-11T23:48:48,269 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - Torch worker started.
2022-10-11T23:48:48,269 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - Python runtime: 3.8.13
2022-10-11T23:48:48,269 [DEBUG] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - W-9000-metrics_model_1.0 State change WORKER_STOPPED -> WORKER_STARTED
2022-10-11T23:48:48,269 [INFO ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - Connecting to: /tmp/.ts.sock.9000
2022-10-11T23:48:48,271 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - Connection accepted: /tmp/.ts.sock.9000.
2022-10-11T23:48:48,271 [INFO ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - Flushing req. to backend at: 1665532128271
2022-10-11T23:48:48,293 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - model_name: metrics_model, batchSize: 1
2022-10-11T23:48:48,299 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - Successfully loaded /home/ubuntu/serve/frontend/server/src/test/resources/metrics_default.yaml.
2022-10-11T23:48:48,299 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - Backend worker process died.
2022-10-11T23:48:48,299 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG - Traceback (most recent call last):
2022-10-11T23:48:48,300 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_service_worker.py", line 223, in <module>
2022-10-11T23:48:48,300 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     worker.run_server()
2022-10-11T23:48:48,300 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_service_worker.py", line 191, in run_server
2022-10-11T23:48:48,300 [INFO ] epollEventLoopGroup-5-8 org.pytorch.serve.wlm.WorkerThread - 9000 Worker disconnected. WORKER_STARTED
2022-10-11T23:48:48,300 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     self.handle_connection(cl_socket)
2022-10-11T23:48:48,301 [DEBUG] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - System state is : WORKER_STARTED
2022-10-11T23:48:48,301 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_service_worker.py", line 156, in handle_connection
2022-10-11T23:48:48,301 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     service, result, code = self.load_model(msg)
2022-10-11T23:48:48,301 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_service_worker.py", line 120, in load_model
2022-10-11T23:48:48,301 [DEBUG] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - Backend worker monitoring thread interrupted or backend worker process died.
java.lang.InterruptedException: null
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.reportInterruptAfterWait(AbstractQueuedSynchronizer.java:2056) ~[?:?]
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2133) ~[?:?]
        at java.util.concurrent.ArrayBlockingQueue.poll(ArrayBlockingQueue.java:432) ~[?:?]
        at org.pytorch.serve.wlm.WorkerThread.run(WorkerThread.java:189) [model-server.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
2022-10-11T23:48:48,301 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     service = model_loader.load(
2022-10-11T23:48:48,301 [WARN ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.BatchAggregator - Load model failed: metrics_model, error: Worker died.
2022-10-11T23:48:48,302 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_loader.py", line 135, in load
2022-10-11T23:48:48,302 [DEBUG] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - W-9000-metrics_model_1.0 State change WORKER_STARTED -> WORKER_STOPPED
2022-10-11T23:48:48,302 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     entry_point, initialize_fn = self._get_class_entry_point(module)
2022-10-11T23:48:48,302 [WARN ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - terminateIOStreams() threadName=W-9000-metrics_model_1.0-stderr
2022-10-11T23:48:48,302 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -   File "/opt/conda/envs/serve/lib/python3.8/site-packages/ts/model_loader.py", line 197, in _get_class_entry_point
2022-10-11T23:48:48,302 [WARN ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - terminateIOStreams() threadName=W-9000-metrics_model_1.0-stdout
2022-10-11T23:48:48,302 [INFO ] W-9000-metrics_model_1.0-stdout MODEL_LOG -     raise ValueError(
2022-10-11T23:48:48,302 [INFO ] W-9000-metrics_model_1.0 org.pytorch.serve.wlm.WorkerThread - Retry worker: 9000 in 21 seconds.
2022-10-11T23:48:48,303 [INFO ] W-9000-metrics_model_1.0-stdout org.pytorch.serve.wlm.WorkerLifeCycle - Stopped Scanner - W-9000-metrics_model_1.0-stdout
2022-10-11T23:48:48,320 [INFO ] W-9000-metrics_model_1.0-stderr org.pytorch.serve.wlm.WorkerLifeCycle - Stopped Scanner - W-9000-metrics_model_1.0-stderr
```